napari/layers/base/base.py

```diff
@@ -130,17 +131,16 @@ def __init__(
         self.scale_factor = 1
 
         self.dims = Dims(ndim)
+
         if scale is None:
-            self._scale = [1] * ndim
-        else:  # covers list and array-like inputs
-            self._scale = list(scale)
+            scale = [1] * ndim
         if translate is None:
-            self._translate = [0] * ndim
-        else:  # covers list and array-like inputs
-            self._translate = list(translate)
-        self._scale_view = np.ones(ndim)
-        self._translate_view = np.zeros(ndim)
-        self._translate_grid = np.zeros(ndim)
+            translate = [0] * ndim
+
+        self._transform = STTransform(scale, translate)
+        self._transform_view = STTransform([1] * ndim, [0] * ndim)
+        self._transform_grid = STTransform([1] * ndim, [0] * ndim)
```

Comment:
(1) why `[1]*` instead of NumPy?
(2) I'm not sure I like combining scale and translate into a single transform rather than chaining them, but we can forget about that for now... ;)
(3) I don't like exactly mirroring the name of the vispy transforms, as it makes it hard for me to tell what you're actually talking about here without going up to see the import. Why not call it ScaleTranslate for example? (The alternative is STTransformModel but I like that even less I think!)

Reply:
I agree with (c)

Reply:
done (1) and (3). I still like ScaleTranslate combined, will comment in main thread on it